adc.c

```diff
@@ -6,9 +6,72 @@
 #define ADMUX_SELECTION_MASK    0x7
 #define MUX5_MASK               0x8
 
+channel_t active_channel = 0;
+
+uint8_t channel_map[VOLTAGE_SENSOR_COUNT];
+
+void setup_adc_channel(uint8_t channel) 
+{
+  // Disable ADC Interrupt
+  bit_false(ADCSRA, 1 << ACIE);
+
+  // Stop ADC conversion
+  bit_false(ADCSRA, 1 << ADSC);
+
+  // Select Channel - ADCSRB, MUX5
+  // If the channel number is higher than 7, the MUX5 bit in ADCSRB needs to be set  
+  if (channel & MUX5_MASK) {
+    bit_true(ADCSRB, (1 << MUX5_BIT_POS));
+  } else {
+    bit_false(ADCSRB, (1 << MUX5_BIT_POS));
+  }
+
+  // Clear the previously selected channel
+  ADMUX &= ~(ADMUX_SELECTION_MASK);
+
+  // Select Channel
+  ADMUX |= (channel & ADMUX_SELECTION_MASK);
+
+  // Enable ADC, Prescaler: 128x
+  ADCSRA = (1 << ADEN) | (1 << ADPS2) | (1 << ADPS1) | (1 << ADPS0);
+
+  // Enable ADC Interrupt
+  bit_true(ADCSRA, 1 << ACIE);
+
+  // Start ADC conversion
+  bit_true(ADCSRA, 1 << ADSC);
+
```

Comment:
I try to use bit_true/false only when I am setting single bits and |= /&= ~ when I'm setting more than one bits or dealing with pre-defined masks. Just my preference since the word bit refers to a single bit. But for the sake of consistency I'll change it.